platformsh-templates/magento2ce
# Refactor python deploy script to php

### Description (*)
Improved Lando Platform.sh plugin compatibility and refactored Python to a PHP script.

### Fixed Issues (if relevant)
* Lando/Platform.sh Plugin incompatible routing
  * Lando did not like the subdirectory proxy defined in the routes.yaml. That's fine—it wasn't being referenced in the deploy script and `/static` is already being handled in the `.platform.app.yaml`
* Refactored Python deploy script to PHP
  * PHP users should be able to easily understand what is going on during their deploy deployments. This makes it much easier to do so.

### Manual testing scenarios (*)
Template Setup
1. Create Magento project from template
   1. The deploy.php script should now output to the deploy log
   2. `platform ssh 'cat .config/installed' should output: `Welcome to Platform.sh! Your Magento site has been deployed!`
   3. Logging in at `/admin` with user `admin` and password `admin123` should require a password reset.

Local Development Testing
1. `platform project:get`
5. `cd {new project dir}`
6. `lando init --recipe platformsh --src cwd --platformsh-site "Magento Project Name"`
7. `lando start`
   1. The deploy.php script should now output to the deploy log
   2. `platform ssh 'cat .config/installed' should output: `Welcome to Platform.sh! Your Magento site has been deployed!`
   3. Logging in at `/admin` with user `admin` and password `admin123` should require a password reset.

### Contribution checklist (*)
